experiment/secret-sync-controller/client/client.go

```diff
@@ -0,0 +1,48 @@
+package client
+
+import (
+	"context"
+	"flag"
+	"os"
+	"path/filepath"
+
+	secretmanager "cloud.google.com/go/secretmanager/apiv1"
+	"google.golang.org/api/option"
+
+	"k8s.io/client-go/kubernetes"
+	_ "k8s.io/client-go/plugin/pkg/client/auth"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+func NewK8sClientset() *kubernetes.Clientset {
+	var kubeconfig *string
+	if home := os.Getenv("HOME"); home != "" {
+		kubeconfig = flag.String("kubeconfig", filepath.Join(home, ".kube", "config"), "(optional) absolute path to the kubeconfig file")
+	} else {
+		kubeconfig = flag.String("kubeconfig", "", "absolute path to the kubeconfig file")
+	}
+	flag.Parse()
+
+	// use the current context in kubeconfig
+	config, err := clientcmd.BuildConfigFromFlags("", *kubeconfig)
+	if err != nil {
+		panic(err.Error())
```

Comment:
In general, don't panic. Instead, return the err and let the caller decide what to do. In this case that means:
- change to `func NewK8sClientset(...) (kubernetes.Interface, error)`
- change err lines like this to `return nil, err`
- change successful lines to `return clientset, nil`

https://golang.org/doc/effective_go.html#panic

Reply:
Done